modules/polaris/src/main/ts/ephox/polaris/words/WordBoundary.ts

```diff
@@ -56,9 +56,16 @@ const isWordBoundary = (map: CharacterMap, index: number): boolean => {
   }
 
   // WB4. Ignore format and extend characters.
-  if (type === ci.EXTEND || type === ci.FORMAT ||
-    prevType === ci.EXTEND || prevType === ci.FORMAT ||
-    nextType === ci.EXTEND || nextType === ci.FORMAT) {
+  if ((type === ci.EXTEND || type === ci.FORMAT) &&
+    (nextType === ci.ALETTER || nextType === ci.NUMERIC || nextType === ci.KATAKANA ||
+      nextType === ci.EXTEND || nextType === ci.FORMAT)
+    ||
+    (nextType === ci.EXTEND ||
+      // TINY-9654: Only ignore format characters if they do not precede a word boundary. Since some format characters overlap with whitespace characters (ex: \ufeff) and
+      // our word extraction logic excludes whitespace characters, if a whitespace-overlapping format character that precedes a word boundary is not split on, whichever word
+      // it is a part of will not be added to the list of extracted words, causing inaccuracies.
+      nextType === ci.FORMAT && (nextNextType === ci.ALETTER || nextNextType === ci.NUMERIC || nextNextType === ci.KATAKANA || nextNextType === ci.EXTEND || nextNextType === ci.FORMAT))
+      && (type === ci.ALETTER || type === ci.NUMERIC || type === ci.KATAKANA || type === ci.EXTEND || type === ci.FORMAT)) {
```

Comment:
This is a bit complex to review in a PR so I'm going to trust that we have automated tests for this file, and none of them broke (other than `'twas` which is an improvement) so this doesn't regress anything.

Reply:
The `'twas` change is a collateral improvement after switching to Polaris's `getWords` API. 

The above change with the word boundary logic for extend and format characters addresses a bug where while "aa`\ufeff`bb" is considered a word by `getWords`, while "a`\ufeff`b" is not considered a word. Format characters like `\ufeff` were supposed to be ignored, so "a`\ufeff`b" should be 1 word.

The bug was uncovered after I removed the obsolete TINY-1166 fix that stripped `\ufeff` characters in `getWords` (`Identify.words` requires `\ufeff` characters to be retained as per https://github.com/tinymce/tinymce/blob/d6e308d85ec9b1bf9285b11f5bb6d0cf19ed0e95/modules/robin/src/test/ts/browser/zone/EntityTest.ts#L24)